stac_fastapi/pgstac/db.py

```diff
@@ -47,33 +46,46 @@ async def con_init(conn):
 ConnectionGetter = Callable[[Request, Literal["r", "w"]], AsyncIterator[Connection]]
 
 
+async def _create_pool(settings: PostgresSettings) -> Pool:
+    """Create a connection pool."""
+    return await asyncpg.create_pool(
+        settings.connection_string,
+        min_size=settings.db_min_conn_size,
+        max_size=settings.db_max_conn_size,
+        max_queries=settings.db_max_queries,
+        max_inactive_connection_lifetime=settings.db_max_inactive_conn_lifetime,
+        init=con_init,
+        server_settings=settings.server_settings.model_dump(),
+    )
+
+
 async def connect_to_db(
     app: FastAPI,
     get_conn: Optional[ConnectionGetter] = None,
     postgres_settings: Optional[PostgresSettings] = None,
+    add_write_connection_pool: bool = False,
+    write_postgres_settings: Optional[PostgresSettings] = None,
 ) -> None:
     """Create connection pools & connection retriever on application."""
-    app_settings = app.state.settings
-
     if not postgres_settings:
         postgres_settings = PostgresSettings()
 
-    if app_settings.testing:
-        readpool = writepool = postgres_settings.testing_connection_string
-    else:
-        readpool = postgres_settings.reader_connection_string
-        writepool = postgres_settings.writer_connection_string
+    app.state.readpool = await _create_pool(postgres_settings)
+
+    if add_write_connection_pool:
+        if not write_postgres_settings:
+            write_postgres_settings = postgres_settings
```

Comment:
by default we will re-use postgres_settings (either defined by the user or from env)

Reply:
Let's see if we ever get feedback on this but technically a user wanting to have the configuration available from env variable do something like `write_postgres_settings=PostgresSettings(_env_prefix="WRITER_")`